src/mrsimulator/spin_system/isotope.py

```diff
@@ -42,57 +51,80 @@ class Isotope(BaseModel):
     """
 
     symbol: str
+    spin_multiplicity: int = Field(default=2, ge=2)
+    gyromagnetic_ratio: float = 0
+    quadrupole_moment: float = 0
+    natural_abundance: float = Field(default=100, ge=0, le=100)
+    atomic_number: int = Field(default=0)
+
     test_vars: ClassVar[Dict] = {"symbol": "1H"}
+    custom_isotope_data: ClassVar[Dict] = {}
 
     class Config:
         extra = "forbid"
         validate_assignment = True
+        allow_mutation = False
+
+    def __init__(self, **kwargs):
+        symbol = kwargs["symbol"]
+        if symbol not in get_all_isotope_symbols():
+            raise Exception(f"Isotope symbol `{symbol}` not recognized.")
```

Comment:
This method of checking for the symbol beforehand disallows parsing in a dictionary from a custom isotope. 

Minimally working example:

```
from mrsimulator import Site
from mrsimulator.spin_system.isotope import Isotope

Isotope.register("custom", gyromagnetic_ratio=123)
site = Site(isotope="custom")
json_dict = site.json()  # Save site json

# de-register the custom isotope data (new Python session)
Isotope.custom_isotope_data = {}

# Attempt to parse in the stored Site json
Site.parse_dict_with_units(json_dict)

---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
...
     69 symbol = kwargs["symbol"]
     70 if symbol not in get_all_isotope_symbols():
---> 71     raise Exception(f"Isotope symbol `{symbol}` not recognized.")
     72 kwargs_new = get_isotope_dict(symbol)
     73 for k, v in kwargs.items():

Exception: Isotope symbol `custom` not recognized.
```

I think the instantiation stuff is complicated enough that we should wait and discuss at group meeting before making any decisions. Can also make a flow-chart for how isotopes can be parsed.